### PR TITLE
VideoPress: Improve attachment details display

### PR DESCRIPTION
**Pull request by obenland**
Status | Before | After
--- | --- | ---
Public | Screenshot 2026-01-06 at 10 23 49 AM | Screenshot 2026-01-06 at 10 23 58 AM
Private | Screenshot 2026-01-06 at 10 12 17 AM | Screenshot 2026-01-06 at 10 23 58 AM

## Proposed changes:
* Refactor from HEREDOC HTML to PHP templating for better maintainability
* Add i18n support with `esc_html_e()` for translatable strings
* Improve output escaping throughout using `esc_attr()`, `esc_url()`
* Use WordPress core CSS classes (`post-attributes-label-wrapper`, `widefat`) for consistent styling
* Fetch playback JWT token via AJAX to display poster for private videos
* Add early return when `guid` is empty
* Make poster image responsive (100% width)

### Does this pull request change what data or activity we track or use?

No

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

1. Upload a video to the Media Library
2. Wait for VideoPress to process the video
3. Click on the video to open the attachment details
4. Verify the VideoPress info section displays:
   - Shortcode input field with proper label
   - URL link to the video
   - Poster image
5. Test with a private video to verify the poster loads after fetching the token

### CODE REVIEW

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 1]
```diff
@@ -241,47 +241,40 @@ public function videopress_information_box( $post ) {
...
+		// If the video is private, we need a metadata token to view the poster, which we don't have access to from here.
+		if ( ! $info->is_private ) :
+			?>
+			<p class="post-attributes-label-wrapper">
+				<label class="post-attributes-label"><?php esc_html_e( 'Poster', 'jetpack' ); ?></label>
+			</p>
+			<?php if ( ! empty( $info->poster ) ) : ?>
+				<img src="<?php echo esc_url( $info->poster ); ?>" width="100%" alt="" />
+			<?php else : ?>
+				<em><?php esc_html_e( 'Processing&hellip;', 'jetpack' ); ?></em>
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php]: Fixed in [fbc4562](https://github.com/Automattic/jetpack/pull/46458/commits/fbc45629996fd7bf1e1dfca1306df4c8b3374deb).

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 2]
```diff
@@ -241,47 +241,40 @@ public function videopress_information_box( $post ) {
...
+
+		<?php
+		// If the video is private, we need a metadata token to view the poster, which we don't have access to from here.
+		if ( ! $info->is_private ) :
+			?>
+			<p class="post-attributes-label-wrapper">
+				<label class="post-attributes-label"><?php esc_html_e( 'Poster', 'jetpack' ); ?></label>
+			</p>
+			<?php if ( ! empty( $info->poster ) ) : ?>
+				<img src="<?php echo esc_url( $info->poster ); ?>" width="100%" alt="" />
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php]: Fixed in [fbc4562](https://github.com/Automattic/jetpack/pull/46458/commits/fbc45629996fd7bf1e1dfca1306df4c8b3374deb).

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 3]
```diff
@@ -239,49 +240,43 @@ public function fields_to_edit( $fields, $post ) {
...
-</div>
-<div class="misc-pub-section misc-pub-poster">
-	<strong>Poster</strong>
-	{$poster}
-</div>
-HTML;
-
-		echo $html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Variables built above.
+		// If the video is private, we need a metadata token to view the poster, which we don't have access to from here.
+		$is_public = VIDEOPRESS_PRIVACY::IS_PUBLIC === $info->privacy_setting || ( VIDEOPRESS_PRIVACY::SITE_DEFAULT === $info->privacy_setting && ! ( new Status() )->is_private_site() );
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:253]: We have preflight checks earlier in the function.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 4]
```diff
@@ -239,49 +240,43 @@ public function fields_to_edit( $fields, $post ) {
...
+				<label class="post-attributes-label"><?php esc_html_e( 'Poster', 'jetpack' ); ?></label>
+			</p>
+			<?php if ( ! empty( $info->poster ) ) : ?>
+				<img src="<?php echo esc_url( $info->poster ); ?>" width="100%" alt="<?php echo esc_attr( $alt_text ); ?>" />
+			<?php else : ?>
+				<em><?php esc_html_e( 'Processing…', 'jetpack' ); ?></em>
+				<?php
+			endif;
+		endif;
 	}
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php]: The suggested change would cause a fatal error. I guess it comes down to style? Open to changing it, but this is fine for now.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 5]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
+		$is_public = VIDEOPRESS_PRIVACY::IS_PUBLIC === $info->privacy_setting || ( VIDEOPRESS_PRIVACY::SITE_DEFAULT === $info->privacy_setting && ! ( new Status() )->is_private_site() );
+		/* Translators: %s is the video title */
+		$alt_text = sprintf( __( 'Poster image for video: %s', 'jetpack' ), get_the_title( $post_id ) );
+		?>
+
+		<p class="post-attributes-label-wrapper">
+			<label class="post-attributes-label" for="videopress-shortcode"><?php esc_html_e( 'Shortcode', 'jetpack' ); ?></label>
+		</p>
+		<input type="text" class="widefat" id="videopress-shortcode" readonly="readonly" value="<?php echo esc_attr( "[videopress $guid]" ); ?>" onclick="this.focus();this.select();" />
+
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:262]: Legacy, I'm not going to change that.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 6]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
+						return;
+					}
+
+					fetch( window.videopressAjax.ajaxUrl, {
+						method: 'POST',
+						credentials: 'same-origin',
+						body: new URLSearchParams( {
+							action: 'videopress-get-playback-jwt',
+							guid: img.dataset.guid,
+							post_id: window.videopressAjax.post_id || 0
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:299]: Happens somewhere else

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 7]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
+					var loading = img ? img.nextElementSibling : null;
+					if ( ! img || ! loading || ! window.videopressAjax ) {
+						return;
+					}
+
+					fetch( window.videopressAjax.ajaxUrl, {
+						method: 'POST',
+						credentials: 'same-origin',
+						body: new URLSearchParams( {
+							action: 'videopress-get-playback-jwt',
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:297]: This already exists

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 8]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
+							loading.style.display = 'none';
+						} else {
+							loading.textContent = loading.dataset.error;
+						}
+					} )
+					.catch( function() {
+						loading.textContent = loading.dataset.error;
+					} );
+				} )();
+				</script>
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:316]: It's fine to be a maintenance concern for this one

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 9]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
-		$guid = get_post_meta( $post_id, 'videopress_guid', true );
+		$guid    = get_post_meta( $post_id, 'videopress_guid', true );
 
 		// If this has not been processed by videopress, we can skip the rest.
-		if ( ! is_videopress_attachment( $post_id ) ) {
+		if ( ! is_videopress_attachment( $post_id ) || empty( $guid ) ) {
 			return;
 		}
 
+		$meta = wp_get_attachment_metadata( $post_id );
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:250]: This is covered with the `is_videopress_attachment` check.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:299]: `videopress-get-playback-jwt` doesn't do nonce checks.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php [hunk 10]
```diff
@@ -239,49 +240,85 @@ public function fields_to_edit( $fields, $post ) {
...
+		</p>
+		<input type="text" class="widefat" id="videopress-shortcode" readonly="readonly" value="<?php echo esc_attr( "[videopress $guid]" ); ?>" onclick="this.focus();this.select();" />
+
+		<p class="post-attributes-label-wrapper">
+			<label class="post-attributes-label"><?php esc_html_e( 'URL', 'jetpack' ); ?></label>
+		</p>
+		<?php printf( '<a href="%1$s">%1$s</a>', esc_url( videopress_build_url( $guid ) ) ); ?>
+
+		<p class="post-attributes-label-wrapper">
+			<label class="post-attributes-label"><?php esc_html_e( 'Poster', 'jetpack' ); ?></label>
```
[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:269]: It's for styling.

[the remark this answers is not in the document]
Comment (obenland) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:269]: It's for styling.

File: projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulopmt1) [reply projects/plugins/jetpack/modules/videopress/class.videopress-edit-attachment.php:253]: Thanks for checking that, I was about to share the same feedback, but it seems it doesn't worth running an AI review locally anymore if we already have this one 😅

**Review by paulopmt1 — APPROVED** [review]
It's working as expected for both public and private videos. Thanks for fixing and improving it @obenland!
